Source/AudioAnalysisTools/Private/Analyzers/BeatDetection.cpp

```diff
@@ -14,13 +14,23 @@ UBeatDetection::UBeatDetection()
 UBeatDetection* UBeatDetection::CreateBeatDetection(int32 InFFTSubbandsSize, int32 InEnergyHistorySize)
 {
 	UBeatDetection* BeatDetection{NewObject<UBeatDetection>()};
+	
+	// We'll set this here, so we only resize the energy history arrays once, in the UpdateFFTSubbandsSize function
+	BeatDetection->EnergyHistorySize = InEnergyHistorySize;
 	BeatDetection->UpdateFFTSubbandsSize(InFFTSubbandsSize);
-	BeatDetection->UpdateEnergyHistorySize(InEnergyHistorySize);
 	return BeatDetection;
 }
 
 void UBeatDetection::UpdateFFTSubbandsSize(int32 InFFTSubbandsSize)
 {
+	// We'll assume nothing, and make sure our user has made a reasonable request
+	if (InFFTSubbandsSize <= 0)
+	{
+		// Tell the user that they've tried to use an incorrect value, and where they tried it
+		UE_LOG(LogAudioAnalysis, Log, TEXT("Beat Detection FFT subbands size '%d' is invalid"), InFFTSubbandsSize);
```

Comment:
Maybe it would be more informative to add "value %d remains" or something like that.